src/proc/namer.rs

```diff
@@ -102,7 +104,12 @@ impl Namer {
             }
             None => {
                 let mut suffixed = base.to_string();
-                if base.ends_with(char::is_numeric) || self.keywords.contains(base.as_ref()) {
+                if base.ends_with(char::is_numeric)
+                    || self.keywords.contains(base.as_ref())
+                    || self
+                        .keywords_case_insensitive
+                        .contains(&base.as_ref().into())
+                {
```

Comment:
I think we can use [`eq_ignore_ascii_case`](https://doc.rust-lang.org/std/primitive.str.html#method.eq_ignore_ascii_case) instead of pulling in a new dep.

Reply:
I'm not sure how you'd use that with a hash set. Or would it be fine to just use a slice and iterate over it?

Reply:
Right, I think what we need is just a wrapper type and a hash impl for it (similar to [this one](https://github.com/seanmonstar/unicase/blob/master/src/ascii.rs#L124-L131)).

```rust
/// A string wrapper type with an ascii case insensitive Eq and Hash impl
struct AsciiUniCase<S: AsRef<str>>(S);

impl<S: AsRef<str>> PartialEq<Self> for AsciiUniCase<S> {
    #[inline]
    fn eq(&self, other: &Self) -> bool {
        self.0.as_ref().eq_ignore_ascii_case(other.0.as_ref())
    }
}

impl<S: AsRef<str>> Eq for AsciiUniCase<S> {}

impl<S: AsRef<str>> Hash for AsciiUniCase<S> {
    #[inline]
    fn hash<H: Hasher>(&self, hasher: &mut H) {
        for byte in self.0.as_ref().as_bytes().iter().map(|b| b.to_ascii_lowercase()) {
            hasher.write_u8(byte);
        }
    }
}
```

plus a debug assert to make sure the reserved words are ascii.

```rust
debug_assert!(reserved_keywords_case_insensitive.iter().all(|s|s.is_ascii()));
```